tests/testmacro.nim

```diff
@@ -123,6 +123,24 @@ suite "Macro transformations test suite":
     macroAsync2(testMacro2, seq, Opt, Result, OpenObject, cstring)
     check waitFor(testMacro2()).len == 0
 
+  test "Future with generics":
+    proc gen(T: typedesc): Future[T] {.async.} =
+      proc testproc(): Future[T] {.async.} =
+        when T is void:
+          return
+        else:
+          return default(T)
+      await testproc()
+
+    waitFor gen(void)
+    discard waitFor gen(int)
```

Comment:
We testing implicit return value with `discard`?

Reply:
So check it for `default(t) == waitFor gen(int)`.

Reply:
Maybe with some crazy optimization this check would be removed?

Reply:
77219d5b1aac7f581bc1e9f19b4a83c44bb90aae